apps/experiments/models.py

```diff
@@ -396,15 +453,50 @@ class Experiment(BaseTeamModel):
     )
     use_processor_bot_voice = models.BooleanField(default=False)
 
+    # Versioning fields
+    working_version = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="versions",
+    )
+    version_number = models.PositiveIntegerField(default=1)
+    is_default_version = models.BooleanField(default=False)
+    is_archived = models.BooleanField(default=False)
+    objects = ExperimentObjectManager()
+
     class Meta:
         ordering = ["name"]
         permissions = [
             ("invite_participants", "Invite experiment participants"),
             ("download_chats", "Download experiment chats"),
         ]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["is_default_version", "working_version"],
+                condition=Q(is_default_version=True),
+                name="unique_default_version_per_experiment",
+            ),
+            models.UniqueConstraint(
+                fields=["version_number", "working_version"],
+                condition=Q(working_version__isnull=False),
+                name="unique_version_number_per_experiment",
+            ),
```

Comment:
Unless I'm missing something, what would be the benefit of doing that? If we're going to add a unique constraint using the id of the working experiment (in the case where working_version_id is None) and the version_number / is_default_version, it seems redundant since the id itself is unique, which itself guarentees uniqueness of the combination?

Reply:
Is it not the case now that you could make the working version 'default' and also one of the versions. (I know there's a restriction on this in code)

Also version number could be duplicated between the working version and the versions.

Reply:
Chatted offline:

* added a check constraint to prevent the working version from also being the default version
* the 2nd constraint is fine because it only applies to versions, not the working copy.